src/hooks/useQuota/useQuota.tsx

```diff
@@ -0,0 +1,143 @@
+import { useState, useContext, useCallback, useEffect } from "react";
+import { Sentry } from "../../utils/errorTracking";
+import { getQuota, NotEligibleError } from "../../services/quota";
+import { ProductContext } from "../../context/products";
+import { transform } from "lodash";
+import { Quota, CampaignPolicy } from "../../types";
+import { usePrevious } from "../usePrevious";
+
+type QuotaState = "DEFAULT" | "FETCHING_QUOTA" | "NO_QUOTA" | "NOT_ELIGIBLE";
+
+export type QuotaHook = {
+  quotaState: string;
+  quotaResponse: Quota | null;
+  allQuotaResponse: Quota | null;
+  updateQuota: () => void;
+  quotaError?: Error;
+};
+
+const filterQuotaWithAvailableProducts = (
+  quota: Quota,
+  products: CampaignPolicy[]
+): Quota => {
+  const filteredQuota: Quota = {
+    remainingQuota: []
+  };
+  transform(
+    quota.remainingQuota,
+    (result: Quota, itemQuota) => {
+      if (products.some(policy => policy.category === itemQuota.category))
+        result.remainingQuota.push(itemQuota);
+    },
+    filteredQuota
+  );
+
+  if (quota.globalQuota) {
+    filteredQuota.globalQuota = [];
+    transform(
+      quota.globalQuota!,
```

Comment:
Noticed we're asserting the existence of this, why is this necessary?

Reply:
This was done because `quota.globalQuota` was declared as a `partial` type. I think we should move forward with declaring both `quota.globalQuota` and `quota.localQuota` as non-partial, since they are always returned.

Let me know what you think.

Reply:
After discussion, I have moved forward with declaring `quota.globalQuota`  and `quota.localQuota` as non-partial.